### ISSUE TITLE
About parameters for different feature extractors 

### ISSUE THREAD

**Issue by wuxchcandoit** [HHHedo/IBMIL#7]
Hi! Thanks for your great work!
May I ask you a question about models' parameters for different feature extractors?
In the paper, different feature extractors are used to extract patches' features. I wonder if the parameters remain unchanged for the same  model because related details are not mentioned in the paper and the supplementary materials. For example, in model `DSMIL`, if the learning rate remains 1e-4, weight dacay remains 5e-3, and the betas in Adam optimizer remains (0.5, 0.9) for feature extractors `Resnet-18 ImageNet pretrained`, `CTransPath SRCL`, and `ViT MoCo V3`? After experiments, I observed that using the parameters given by the code and feature extractor `CTransPath SRCL`, DSMIL can only get about 66% acc and 67% auc in the dataset Camelyon16.

**Comment by HHHedo** [comment]
Hi,The parameters of feature extractor is not updated due to the hardware constraints, since the bag length of a WSI can be extremely large.As mentioned in paper, the bag-level MIL is usually in a two-stage manner instead of end-to-end, which means training the feature extractor first and then training the aggregator with the feature extractor fixed.在 2023年9月28日，14:53，wuxchcandoit ***@***.***> 写道：﻿
Hi! Thanks for your great work!
May I ask you a question about models' parameters for different feature extractors?
In the paper, different feature extractors are used to extract patches' features. I wonder if the parameters remain unchanged for the same  model because related details are not mentioned in the paper and the supplementary materials. For example, in model DSMIL, if the learning rate remains 1e-4, weight dacay remains 5e-3, and the betas in Adam optimizer remains (0.5, 0.9) for feature extractors Resnet-18 ImageNet pretrained, CTransPath SRCL, and ViT MoCo V3? After experiments, I observed that using the parameters given by the code and feature extractor CTransPath SRCL, DSMIL can only get about 66% acc and 67% auc in the dataset Camelyon16.

**Comment by wuxchcandoit** [comment]
Thanks for your reply. Please let me explain my question again. I wonder if the MIL models' parameters remain unchanged for the same MIL model like `DSMIL`, `ABMIL`, and `TransMIL` after using different feature extractors to extract patches' features?

**Comment by HHHedo** [comment]
Hi,For your question, yes.The released code is based on Dsmil. If you are familiar with the code base, you will find the extracted feature will be stored offline. 在 2023年9月28日，15:12，wuxchcandoit ***@***.***> 写道：﻿
Thanks for your reply. Please let me explain my question again. I wonder if the parameters remain unchanged for the same MIL model like DSMIL, 'ABMIL', and TransMIL after using different feature extractors to extract patches' features?

**Comment by wuxchcandoit** [comment]
Yes. I know that the extracted features will be stored offline. But the focus of my problem is not on the parameters of the feature extractors. The focus of my problem is on the parameters of MIL models. Because I found that using the default parameters of DSMIL fails to get the ideal results for different extracted features. For example, using the parameters given by the code and extracted features from feature extractor `CTransPath SRCL`, `DSMIL` can only get about 66% acc and 67% auc in the dataset Camelyon16 while `DSMIL` can get similar results in paper using extracted features from feature extractor `ResNet-18 ImageNet pretrained`.

**Comment by HHHedo** [comment]
Hi, I am confused about the “parameter”. According to your first question, I guess it means the training config like learning rate?在 2023年9月28日，15:25，wuxchcandoit ***@***.***> 写道：﻿
Yes. I know that the extracted feature will be stored offline. But the focus of my problem is not on the parameters of the feature extractors. The focus of my problem is on the parameters of MIL models. Because I found that using the default parameters of DSMIL fails to get the ideal results for different extracted features. For example, using the parameters given by the code and extracted features from feature extractor CTransPath SRCL, DSMIL can only get about 66% acc and 67% auc in the dataset Camelyon16 while DSMIL cane get similar results in paper using extracted features from feature extractor ResNet-18 ImageNet pretrained

**Comment by wuxchcandoit** [comment]
Yes!! like learning rate, weight decay, the betas in optimizer in MIL models.

**Comment by HHHedo** [comment]
For different features, we use the same default training configurations according to their official code. For the poor performance you achieved, I guess something wrong happens in extracting features since the effectiveness of SGCL has been proved in many papers.在 2023年9月28日，15:37，wuxchcandoit ***@***.***> 写道：﻿
Yes!! like learning rate, weight decay, the betas in optimizer in MIL models.

**Comment by wuxchcandoit** [comment]
I see! I guess maybe the ways we pre-process the dataset Camelyon16 are different. I used the data pre-processing in [dsmil](https://github.com/binli123/dsmil-wsi/) while you used the processed C16 in your previous work (which is mainly based on this [repo](https://github.com/deroneriksson/python-wsi-preprocessing)). Therefore, the default parameters fail to work and I fail to get the ideal results. That is a touchy issue because it is time-consuming to pre-process the data again from scratch. Anyway, thanks you so much for replying my questions!!! sincerely.
